argmin/src/solver/conjugategradient/cg.rs

```diff
@@ -89,7 +89,7 @@ where
     }
 }
 
-impl<P, O, F> Solver<O, IterState<P, (), (), (), F>> for ConjugateGradient<P, F>
+impl<P, O, F, R> Solver<O, IterState<P, (), (), (), F, R>> for ConjugateGradient<P, F>
```

Comment:
CG actually has residuals. Updating this to use the residual in `IterState` could be a good additional test bed, but it can also be skipped for this PR.